tenseal/cpp/tensors/BUILD

```diff
@@ -16,9 +17,11 @@ cc_library(
     hdrs = [
         "api.h",
         "bfvvector.h",
+        "bfvtensor.h",
         "ckkstensor.h",
         "ckksvector.h",
         "encrypted_tensor.h",
+        "encrypted_tensor_bfv.h",
```

Comment:
encrypted_tensor should be a generic interface for all tensors in TenSEAL? Why do we need encrypted_tensor_bfv too?

Reply:
Yes you are right of course, sorry.
I just got a little confused with the _scale_ method in _encrypted_tensor.h_ given that it is not supposed to be used in BFV and indeed only now I've discovered that it's declared as not implemented for BFV (for the BFVVector counterpart I mean).
Also the same for _polyval_inplace_ method which was using double so I initially changed the data type to the template _plain_data_t_ but it was messing with BFVVector which was using double as template but the method itself was again declared as not implemented, so I did the same now for BFVTensor.

Sorry for the confusion